per/views.py

```diff
@@ -1,70 +1,34 @@
-from api.views import (
-    bad_request,
-    bad_http_request,
-    PublicJsonPostView,
-    PublicJsonRequestView,
-)
-from rest_framework import viewsets
-from django.contrib.auth.models import User
+from django.db import transaction
+from django.http import JsonResponse
+from django.utils import timezone
+from rest_framework import authentication, permissions
+from rest_framework.views import APIView
 from .models import (
     Draft, Form, FormData, WorkPlan, Overview
 )
-from rest_framework.authtoken.models import Token
-
-def create_draft(raw):
-    Draft.objects.filter(code=raw['code'], country_id=raw['country_id'], user_id=raw['user_id']).delete()  # If exists (a previous draft), delete it.
-    draft = Draft.objects.create(code       = raw['code'],
-                               user_id      = raw['user_id'],
-                               data         = raw['data'],
-                               country_id   = raw['country_id'],
-                               )
-    draft.save()
-    return draft
-
-def create_form(raw):
-
-    form = Form.objects.create(code         = raw['code'],
-                               name         = raw['name'],
-                               language     = raw['language'],
-                               user_id      = raw['user_id'],
-                               country_id   = raw['country_id'],
-                               ns           = raw['ns'],
-                               ip_address   = raw['ip_address'],
-                               started_at   = raw['started_at'],
-                               ended_at     = raw['ended_at'],
-                               submitted_at = raw['submitted_at'],
-                               comment      = raw['comment'],
-                               validated    = raw['validated'],
-                               finalized    = raw['finalized'],
-                               # unique_id  = raw['unique_id'], # only KoBo form provided
-                               )
-    form.save()
-    return form
-
-def create_form_data(raw, form):
-    form_data = FormData.objects.create(form= form,
-                            question_id     = raw['id'],
-                            selected_option = raw['op'],
-                            notes           = raw['nt'],
-                            )
-    form_data.save()
-    return form_data
-
-class FormSent(PublicJsonPostView):
-    def handle_post(self, request, *args, **kwargs):
-
-        #if not request.user.is_authenticated:
-        #    return bad_request('Could not insert PER data due to not logged in user.')
-
-        body = json.loads(request.body.decode('utf-8'))
-
-        x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR')
-        if x_forwarded_for:
-            body['ip_address'] = x_forwarded_for.split(',')[-1].strip()
-        else:
-            body['ip_address'] = request.META.get('REMOTE_ADDR')
+from api.views import bad_request
+
+
+def get_client_ip(request):
```

Comment:
@batpad @thenav56 No clue if there's a nicer way to achieve this so I just made this a function from Zoltán's code. I don't even know why are we saving IPs here...